themes/finna2/js/js-helper.js

```diff
@@ -0,0 +1,250 @@
+/* exported jsHelper */
+var jsHelper = (function jsHelper() {
+
+  /**
+   * Create a html element
+   * 
+   * @param {String} tag 
+   * @param {String} className 
+   * @param {Object} attributes 
+   */
+  function createElement(tag, className, attributes) {
+    var element = document.createElement(tag);
+    element.className = className;
+    if (attributes) {
+      for (var attr in attributes) {
+        if (attributes.hasOwnProperty(attr)) {
+          element.setAttribute(attr, attributes[attr]);
+        }
+      }
+    }
+    return element;
+  }
+
+  /**
+   * Find next sibling with syntax
+   * 
+   * @param {HTMLElement} elem 
+   * @param {String} selector 
+   */
+  function getNextElementSibling(elem, selector) {
+    var sibling = elem.nextElementSibling;
+    while (sibling) {
+      if (sibling.matches(selector)) {
+        return sibling;
+      }
+      sibling = sibling.nextElementSibling;
+    }
+    return undefined;
+  }
+
+  /**
+   * Find previous sibling with syntax
+   * 
+   * @param {HTMLElement} elem 
+   * @param {String} selector 
+   */
+  function getPreviousElementSibling(elem, selector) {
+    var sibling = elem.previousElementSibling;
+    while (sibling) {
+      if (sibling.matches(selector)) {
+        return sibling;
+      }
+      sibling = sibling.previousElementSibling;
+    }
+    return undefined;
+  }
+
+  /**
+   * Binds an event to the element and given selector checks if the event can be ran.
+   * Useful for binding events to dynamically created buttons etc.
+   * 
+   * @param {HTMLElement} el
+   * @param {String} selector
+   * @param {String} eventName
+   * @param {Array} path 
+   */
+  function addDynamicListener(el, selector, eventName, handler) {
+    el.addEventListener(eventName, function checkEvent(e) {
+      // loop parent nodes from the target to the delegation node
+      for (var target = e.target; target && target !== this; target = target.parentNode) {
+        if (target.matches(selector)) {
+          handler.call(target, e);
+          break;
+        }
+      }
+    }, false);
+  }
+
+  /**
+   * Find all parents of element until selector is satisfied.
```

Comment:
Tätä voisi vähän tarkentaa: `Return all parents of an element up to and including the first one that matches the given selector.`